src/utils/log2_fold_change.rs

```diff
@@ -0,0 +1,49 @@
+use std::ops::Not;
+
+use ordered_float::NotNan;
+
+use crate::variants::model::AlleleFreq;
+use crate::utils::comparison::Comparison;
+
+#[derive(Debug, Clone)]
+pub(crate) struct Log2FoldChange {
+    vaf_a: AlleleFreq,
+    vaf_b: AlleleFreq,
+    l2fc: f64,
+}
+
+impl Log2FoldChange {
+    pub(crate) fn new(vaf_a: AlleleFreq, vaf_b: AlleleFreq) -> Self {
+        Self {
+            vaf_a,
+            vaf_b,
+            l2fc: vaf_a.log2() - vaf_b.log2(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub(crate) struct Log2FoldChangePredicate(pub(crate) Comparison, pub(crate) NotNan<f64>);
+
+impl Log2FoldChangePredicate {
+    pub(crate) fn is_true(&self, lfc: &Log2FoldChange) -> bool {
+        // FIXME: equality checks should probably allow for some error
+        let v = *self.1;
+        match self.0 {
+            Comparison::Equal => lfc.l2fc == v,
```

Comment:
For the strict comparison lints, there should indeed be taken seriously here. Just use `relative_eq!` from the approx crate (already a dependency).